src/components/Pages/Admin/ToolbarEditor.vue

```diff
@@ -0,0 +1,461 @@
+                { name: 'Copy_Visual_Style', mode: 35, src: '/vendor/img/tool_icons/Mode_copyvisualstyle.svg' },
+                { name: 'Angle', mode: 36, src: '/vendor/img/tool_icons/Mode_angle.svg' },
+                { name: 'Vector_From_Point', mode: 37, src: '/vendor/img/tool_icons/Mode_vectorfrompoint.svg' },
+                { name: 'Distance', mode: 38, src: '/vendor/img/tool_icons/Mode_distance.svg' },
+                { name: 'Move_Rotate', mode: 39, src: '/vendor/img/tool_icons/Mode_moverotate.svg' },
+                { name: 'Translateview', mode: 40, src: '/vendor/img/tool_icons/Mode_translateview.svg' },
+                { name: 'Zoom_In', mode: 41, src: '/vendor/img/tool_icons/Mode_zoomin.svg' },
+                { name: 'Zoom_Out', mode: 42, src: '/vendor/img/tool_icons/Mode_zoomout.svg' },
+                { name: 'Polar_Diameter', mode: 44, src: '/vendor/img/tool_icons/Mode_polardiameter.svg' },
+                { name: 'Segment_Fixed', mode: 45, src: '/vendor/img/tool_icons/Mode_segmentfixed.svg' },
+                { name: 'Angle_Fixed', mode: 46, src: '/vendor/img/tool_icons/Mode_anglefixed.svg' },
+                { name: 'Locus', mode: 47, src: '/vendor/img/tool_icons/Mode_locus.svg' },
+                { name: 'Macro', mode: 48, src: '/vendor/img/tool_icons/Mode_tool.svg' },
+                { name: 'Area', mode: 49, src: '/vendor/img/tool_icons/Mode_area.svg' },
+                { name: 'Slope', mode: 50, src: '/vendor/img/tool_icons/Mode_slope.svg' },
+                { name: 'Regular_Polygon', mode: 51, src: '/vendor/img/tool_icons/Mode_regularpolygon.svg' },
+                { name: 'Show_Hide_Checkbox', mode: 52, src: '/vendor/img/tool_icons/Mode_showcheckbox.svg' },
+                { name: 'Compasses', mode: 53, src: '/vendor/img/tool_icons/Mode_compasses.svg' },
+                { name: 'Mirror_At_Circle', mode: 54, src: '/vendor/img/tool_icons/Mode_mirroratcircle.svg' },
+                { name: 'Ellipse_Three_Points', mode: 55, src: '/vendor/img/tool_icons/Mode_ellipse3.svg' },
+                { name: 'Hyperbola_Three_Points', mode: 56, src: '/vendor/img/tool_icons/Mode_hyperbola3.svg' },
+                { name: 'Parabola', mode: 57, src: '/vendor/img/tool_icons/Mode_parabola.svg' },
+                { name: 'Fitline', mode: 58, src: '/vendor/img/tool_icons/Mode_fitline.svg' },
+                { name: 'Record_To_Spreadsheet', mode: 59, src: '/vendor/img/tool_icons/Mode_recordtospreadsheet.svg' },
+                { name: 'Attach_Detach_Point', mode: 67, src: '/vendor/img/tool_icons/Mode_attachdetachpoint.svg' },
+            ],
+        };
+    },
+
+    computed: {
+        ...mapGetters('users', {
+            teacher: 'current',
+        }),
+
+        toolbars() {
+            return this.teacher.toolbars;
+        },
+    },
+
+    methods: {
+        ...mapActions('users', {
+            patch: 'patch',
+        }),
+
+        handleChange() {
```

Comment:
The usage of  `handleChange` method is completely against Vue's reactive design. This should be defined as a [watcher](https://vuejs.org/v2/guide/computed.html#Watchers) on the `content` property. As a result, whenever `content` changes, the new `input` event is emitted (and you don't have to manually handle the update procedure and think about the places where the update should be performed). All the methods that modify toolbar list should then modify this property and let Vue do it's reactive magic.

However, this would require rewriting this component, so if there is no budget to do this, it's fine.